### PR TITLE
MISUV-8273 - Update reporting frequency logic - Amend 'Manage your details' page

### PR DESCRIPTION
**Pull request by JordHemsworth**

### CODE REVIEW

File: app/views/incomeSources/manage/ManageIncomeSourceDetails.scala.html [hunk 1]
```diff
@@ -250,8 +250,10 @@
             ))
                 ++ getQuarterlyType
                 ++ {
-            if(viewModel.itsaHasMandatedOrVoluntaryStatusCurrentYear
-                    && viewModel.latencyDetails.isDefined) getTaxYears else Seq.empty
+            viewModel.latencyYearTwoIsQuarterly match {
+                case Some(true) if viewModel.latencyDetails.isDefined => getTaxYears
+                case _ => Seq.empty
+            }
```
Comment (Aashvin) [inline app/views/incomeSources/manage/ManageIncomeSourceDetails.scala.html:256]: Could this be moved out of the view into the view model, just to remove a bit of logic from the view

File: app/models/incomeSourceDetails/viewmodels/ManageIncomeSourceDetailsViewModel.scala [hunk 2]
```diff
@@ -28,7 +28,8 @@ case class ManageIncomeSourceDetailsViewModel(incomeSourceId: IncomeSourceId,
                                               tradingStartDate: Option[LocalDate],
                                               address: Option[AddressModel],
                                               isTraditionalAccountingMethod: Boolean,
-                                              itsaHasMandatedOrVoluntaryStatusCurrentYear: Boolean,
+                                              latencyYearTwoIsQuarterly: Option[Boolean],
+                                              latencyYearOneIsQuarterly: Option[Boolean],
                                               taxYearOneCrystallised: Option[Boolean],
                                               taxYearTwoCrystallised: Option[Boolean],
```
Comment (Aashvin) [inline app/models/incomeSourceDetails/viewmodels/ManageIncomeSourceDetailsViewModel.scala:34]: I've noticed latency years and crystallisation years both have year1 and year2 which seem to be coupled throughout, would it be worth creating case classes for these?

File: test/services/ITSAStatusServiceSpec.scala [hunk 3]
```diff
@@ -39,37 +46,80 @@ class ITSAStatusServiceSpec extends TestSupport with MockITSAStatusConnector {
         setupGetITSAStatusDetail(testNino, yearRange, false, false)(Right(List(successITSAStatusResponseMTDMandatedModel)))
         TestITSAStatusService.hasMandatedOrVoluntaryStatusCurrentYear(testMtdItUser, headerCarrier, ec).futureValue shouldBe true
       }
-      "return false" in {
+      //      "return false" in {
```
Comment (Aashvin) [inline test/services/ITSAStatusServiceSpec.scala:49]: Can these tests be uncommented?

File: app/controllers/manageBusinesses/manage/ManageIncomeSourceDetailsController.scala [hunk 4]
```diff
@@ -186,49 +188,65 @@ class ManageIncomeSourceDetailsController @Inject()(val view: ManageIncomeSource
...
-      }
-    } else {
-      Future(Left(
-        new Error("Unable to find income source")
-      ))
+    desiredIncomeSourceMaybe match {
+      case Some(desiredIncomeSource) =>
+        desiredIncomeSource.latencyDetails match {
+          case Some(latencyDetails) =>
+            itsaStatusService.hasMandatedOrVoluntaryStatusForLatencyYears(Some(latencyDetails)).flatMap {
```
Comment (Aashvin) [inline app/controllers/manageBusinesses/manage/ManageIncomeSourceDetailsController.scala:205]: This has become quite nested, maybe abstract this to it's own function? Or some other way if you think of something better

File: app/controllers/manageBusinesses/manage/ManageIncomeSourceDetailsController.scala [hunk 5]
```diff
@@ -158,7 +158,8 @@ class ManageIncomeSourceDetailsController @Inject()(val view: ManageIncomeSource
       tradingStartDate = incomeSource.tradingStartDate,
       address = incomeSource.address,
       isTraditionalAccountingMethod = incomeSource.cashOrAccruals,
-      itsaHasMandatedOrVoluntaryStatusCurrentYear = itsaStatus,
+      latencyYearTwoIsQuarterly = Some(latencyYearOneStatus),
```
[the remark this answers is not in the document]
Comment (JordHemsworth) [reply app/controllers/manageBusinesses/manage/ManageIncomeSourceDetailsController.scala]: In the old journey

File: test/services/ITSAStatusServiceSpec.scala (hunk 3 above)
Comment (JordHemsworth) [reply test/services/ITSAStatusServiceSpec.scala]: Sorted

**Review by Aashvin — APPROVED** [review]
Great work, it's looking good!

**Pull request merged**